datafusion/core/src/dataframe.rs

```diff
@@ -663,12 +663,7 @@ impl DataFrame {
     ) -> Result<DataFrame> {
         let expr = on_exprs.into_iter().reduce(Expr::and);
         let plan = LogicalPlanBuilder::from(self.plan)
-            .join(
-                right.plan,
-                join_type,
-                (Vec::<Column>::new(), Vec::<Column>::new()),
-                expr,
-            )?
+            .join_on(right.plan, join_type, expr)?
```

Comment:
There are more code can be like this.

like `join_on_false()` .....

Reply:
I was also thinking there are some additional documentation improvements that can be done (namely suggesting people use `join_on` instead of the various other join functions. I can make a PR with such improvements as a follow on, perhaps